### PR TITLE
[SYCLomatic] Update the the value of "-std" option less then C++17 to C++17 in the generated makefile

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/GenMakefile.cpp [hunk 1]
```diff
@@ -242,15 +242,18 @@ static void getCompileInfo(
...
+        printf("####### [%s]\n", Option.c_str());
         size_t Idx = 0;
         for (; Idx < Option.length(); Idx++) {
           if (std::isdigit(Option[Idx]))
             break;
         }
         auto Version = Option.substr(Idx, Option.length() - Idx);
+        printf("####### Version [%s]\n", Version.c_str());
         int Val = std::atoi(Version.c_str());
+        printf("####### Val [%d]\n", Val);
```
Comment (0x12CC) [inline clang/lib/DPCT/GenMakefile.cpp]: Were these `printf` statements added intentionally or are they only needed for debugging?

Comment (tomflinda) [reply clang/lib/DPCT/GenMakefile.cpp]: Only, just for debug purpose, should be removed.

**Review by 0x12CC — APPROVED** [review]

File: clang/lib/DPCT/GenMakefile.cpp [hunk 2]
```diff
@@ -249,8 +249,20 @@ static void getCompileInfo(
...
         int Val = std::atoi(Version.c_str());
+
+        llvm::SmallString<512> RealPath;
+        llvm::sys::fs::real_path(FileName, RealPath, true);
+        if (!llvm::sys::path::is_absolute(FileName))
+          RealPath = Directory + "/" + FileName;
+        makeCanonical(RealPath);
+
+        bool HasCudaSemantics = false;
+        if (IncludeFileMap[std::string(RealPath.str())]) {
```
Comment (zhiweij1) [inline clang/lib/DPCT/GenMakefile.cpp]: IncludeFileMap.count(std::string(RealPath.str())) && IncludeFileMap.at(std::string(RealPath.str()))

Comment (tomflinda) [reply clang/lib/DPCT/GenMakefile.cpp]: ok

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**